repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/valkyrie/CullingWFlexibleDeviceOptionsRemoveTest.groovy

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.valkyrie
+
+import groovy.json.JsonSlurper
+import org.junit.experimental.categories.Category
+import org.openrepose.framework.test.ReposeValveTest
+import org.openrepose.framework.test.mocks.MockIdentityV2Service
+import org.openrepose.framework.test.mocks.MockValkyrie
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import org.rackspace.deproxy.Response
+import scaffold.category.Slow
+import spock.lang.Unroll
+
+import static features.filters.valkyrie.CullingWFlexibleDeviceOptionsTestsHelper.jsonrespbody
+import static features.filters.valkyrie.CullingWFlexibleDeviceOptionsTestsHelper.randomTenant
+
+@Category(Slow)
+class CullingWFlexibleDeviceOptionsRemoveTest extends ReposeValveTest {
+    def static originEndpoint
+    def static identityEndpoint
+    def static valkyrieEndpoint
+
+    static MockIdentityV2Service fakeIdentityService
+    static MockValkyrie fakeValkyrie
+    static Map params = [:]
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        reposeLogSearch.cleanLog()
+
+        params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.cleanConfigDirectory()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie/collectionresources", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie/collectionresources/nullidaction", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie/collectionresources/nullidaction/remove", params)
+        repose.start()
+
+        originEndpoint = deproxy.addEndpoint(properties.targetPort, 'origin service')
+        fakeIdentityService = new MockIdentityV2Service(properties.identityPort, properties.targetPort)
+        identityEndpoint = deproxy.addEndpoint(properties.identityPort, 'identity service', null, fakeIdentityService.handler)
+        fakeIdentityService.checkTokenValid = true
+
+        fakeValkyrie = new MockValkyrie(properties.valkyriePort)
+        valkyrieEndpoint = deproxy.addEndpoint(properties.valkyriePort, 'valkyrie service', null, fakeValkyrie.handler)
+    }
+
+    @Unroll
+    def "Remove - permission: #permission for #method with tenant: #tenantID and deviceIDs: #deviceID, #deviceID2 should return a #responseCode"() {
+        given: "reconfig repose with null uri remove action"
+        "a list of permission defined in valkyrie"
```

Comment:
What's up with this random string literal?

Reply:
It appears to be acting as a comment. I will add `and` blocks to give these strings more meaning and the test more readability.

Reply:
As a side note, the first `given` string implies that we used to reconfigure Repose for these tests which might account for the usage of the `Slow` category, but that clearly does not happen anymore.